### PR TITLE
update(JS): web/javascript/reference/global_objects/string/replace

### PR DESCRIPTION
**Pull request by undead404**
Оригінальний вміст: [String.prototype.replace()@MDN](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Global_Objects/String/replace), [сирці String.prototype.replace()@GitHub](https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/string/replace/index.md)

Нові зміни:
- [mdn/content@cf402f7](https://github.com/mdn/content/commit/cf402f743c3b7f8f4ca954097c76d118119a81a2)
- [mdn/content@750f982](https://github.com/mdn/content/commit/750f982eeb575963117b4a61af26df9ef8036fed)
- [mdn/content@f82233a](https://github.com/mdn/content/commit/f82233a2e58b3804af334ae0318fb8627153474c)
- [mdn/content@8a5ab92](https://github.com/mdn/content/commit/8a5ab92ceab8fe1b8f44597381cf3fd996e252e9)
- [mdn/content@d6f63f6](https://github.com/mdn/content/commit/d6f63f6e41146f459e21046af1c51fadb772de70)

### CODE REVIEW

**Review by AdriandeCita — CHANGES_REQUESTED** [review]
Є свіжі зміни:
- [mdn/content@8a5ab92](https://github.com/mdn/content/commit/8a5ab92ceab8fe1b8f44597381cf3fd996e252e9)

**Comment by undead404** [comment]
Оновлено

**Comment by undead404** [comment]
Без змін

**Comment by AdriandeCita** [comment]
Тут доволі масштабні зміни:
https://github.com/mdn/content/commit/f82233a2e58b3804af334ae0318fb8627153474c#diff-2362e7e0aa3e051b3cdc27e43b5dfa2a9cccc505366083c1430636c52fd09b0b

**Review by AdriandeCita — CHANGES_REQUESTED** [review]
per comment

**Comment by undead404** [comment]
Оновлено:
* https://github.com/mdn/content/commit/f82233a2e58b3804af334ae0318fb8627153474c
* https://github.com/mdn/content/commit/cf402f743c3b7f8f4ca954097c76d118119a81a2

File: files/uk/web/javascript/reference/global_objects/string/replace/index.md [hunk 1]
```diff
@@ -15,114 +15,132 @@ browser-compat: javascript.builtins.String.replace
...
+## Опис
 
-### Повернене значення
+Цей метод не видозмінює рядкове значення, на котрому викликаний. Він повертає новий рядок.
 
-Новий рядок, в якому деякі (або всі) збіги з патерном заміщені переданим замінником.
+Рядковий патерн замінюється лише раз. Аби виконати глобальний пошук із заміною, слід застосувати регулярний вираз із позначкою `g` або використати натомість [`replaceAll()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll).
 
-## Опис
+Якщо `pattern` є об'єктом із методом [`Symbol.replace`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Symbol/replace) (в т.ч. об'єктом `RegExp`), то такий метод викликається з цільовим рядком і `replacement` як аргументами. ого повернене значення стає поверненим значенням `replace()`. У такому випадку поведінка `replace()` цілком захована у методі `@@replace`: наприклад, будь-які згадки "груп захоплення" в описі нижче насправді є функціональністю, наданою [`RegExp.prototype[@@replace]`](/uk/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@replace).
```
Comment (AdriandeCita) [inline files/uk/web/javascript/reference/global_objects/string/replace/index.md:47]:
```suggestion
Якщо `pattern` є об'єктом із методом [`Symbol.replace`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Symbol/replace) (в т.ч. об'єктом `RegExp`), то такий метод викликається з цільовим рядком і `replacement` як аргументами. Його повернене значення стає поверненим значенням `replace()`. У такому випадку поведінка `replace()` цілком захована у методі `@@replace`: наприклад, будь-які згадки "груп захоплення" в описі нижче насправді є функціональністю, наданою [`RegExp.prototype[@@replace]`](/uk/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@replace).
```

File: files/uk/web/javascript/reference/global_objects/string/replace/index.md [hunk 2]
```diff
@@ -15,114 +15,132 @@ browser-compat: javascript.builtins.String.replace
...
+- `offset` (відступ)
+  - : Відступ підрядка, що дав збіг, у межах всього досліджуваного рядка. Наприклад, якщо увесь рядок був `'abcd'`, а підрядок, що дав збіг, був `'bc'`, то цей аргумент буде `1`.
+- `string` (рядок)
+  - : Увесь досліджуваний рядок.
+- `groups` (групи)
+  - : Об'єкт, чиї ключі – використовувані імена груп, і чиї значення – частини, що дали збіг (`undefined`, якщо збігу немає). Присутній лише тоді, коли `pattern` містить принаймні одну іменовану групу захоплення.
 
-(Точна кількість аргументів залежить від того, чи був першим аргументом переданий {{jsxref("RegExp", "регулярний вираз")}}, і кількості вказаних груп в дужках в ньому, якщо був.)
+Конкретне число аргументів залежить від того, чи є перший аргумент об'єктом {{jsxref("RegExp")}} – і, якщо це так, того, скільки груп захоплення він має.
```
Comment (AdriandeCita) [inline files/uk/web/javascript/reference/global_objects/string/replace/index.md:105]: Коми там точно недостатньо
```suggestion
Конкретне число аргументів залежить від того, чи є перший аргумент об'єктом {{jsxref("RegExp")}} – і, якщо це так – того, скільки груп захоплення він має.
```

File: files/uk/web/javascript/reference/global_objects/string/replace/index.md [hunk 3]
```diff
@@ -175,14 +193,49 @@ let newString = propertyName.replace(/[A-Z]/g, '-' + '$&'.toLowerCase()); // т
...
+
+Припустімо, що необхідно створити замінювача, що додає дані про відступ до кожного рядка, котрий дав збіг. У зв'язку з тим, що функція-замінювач отримує параметр `offset`, це буде тривіальним, якщо регулярний вираз статично відомий.
+
+```js
+console.log(
+  'abcd'.replace(/(bc)/, (match, p1, offset) => `${match} (${offset}) `),
+); // "abc (1) d"
+```
+
+Проте такий замінювач буде важко узагальнити, якщо він повинен працювати з будь-яким патерном регулярного виразу. Замінювач є _варіативним_: число аргументів, котре він отримує, залежить від числа присутніх груп захоплення. Можна застосувати [решту параметрів](/uk/docs/Web/JavaScript/Reference/Functions/rest_parameters), але вона так само захопить у масив `offset`, `string` тощо. Факт того, що `groups` може бути або не бути переданим, залежно від ідентичності регулярного виразу, також ускладнює узагальнене з'ясування того, котрий аргумент відповідає `offset`.
```
Comment (AdriandeCita) [inline files/uk/web/javascript/reference/global_objects/string/replace/index.md:214]:
```suggestion
Проте такий замінювач буде важко узагальнити, якщо він повинен працювати з будь-яким патерном регулярного виразу. Замінювач є _варіативним_: число аргументів, котре він отримує, залежить від числа присутніх груп захоплення. Можна застосувати [решту параметрів](/uk/docs/Web/JavaScript/Reference/Functions/rest_parameters), але вона так само захопить у масив `offset`, `string` тощо. Факт того, що `groups` може бути або не бути переданим, залежно від ідентичності регулярного виразу, також ускладнює узагальнене з'ясування того, котрий із аргументів відповідає параметрові `offset`.
```

**Comment by undead404** [comment]
Без змін

**Pull request merged**